hack/test.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+
+# 
+# Runs a test of the knative serving installation by deploying and then 
+# invoking an http echoing server.
+#
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+main() {
+  local em=$(tput bold)$(tput setaf 2)
+  local me=$(tput sgr0)
+
+  # Drop some debug in the event even the above excessive wait does not work.
+  echo "${em}Testing...${me}"
+
+  echo "${em}  Creating echo server${me}"
+  n=0
+  until [ $n -ge 10 ]; do
+    cat <<EOF | kubectl apply -f - && break
+  apiVersion: serving.knative.dev/v1
+  kind: Service
+  metadata:
+    name: echo
+    namespace: func
+  spec:
+    template:
+      spec:
+        containers:
+          - image: docker.io/jmalloc/echo-server
```

Comment:
Not sure if pull limit would be issue here. Maybe it would be better to mirror this in our quay? @jrangelramos knows more about the limitations of docker.io

Reply:
Good point.  I will keep an eye on this as a potential failure mode, and fork if/ when we hit that limit, or better yet replace this test entirely with a Function which does the same thing.